### ISSUE TITLE
Invalid Rar Archive Usage/ Cannot save settings in postprocessing

### ISSUE THREAD

**Issue by Quiksmage** [SickRage/SickRage#3345]
Branch/Commit: master
Commit: 5071e52c498fc145838231f4fb85565fa6ffa70b

OS: Windows Server 2012 R2
What you did: set path to C:\Program Files\WinRAR\UnRAR.exe
What happened: Invalid Rar Archive
What you expected: Successful unrar
Logs:
```
Failed to extract the archive black.sails.s04e06.1080p.hdtv.x264-brisk.rar: Invalid Rar Archive Usage
```

I tried to set it back to the way it was as "unrar" along with adding the path C:\Program Files\WinRAR\ to my system variables, but for whatever reason, I can no longer save in this screen. My settings will always revert on a refresh. 

So, there's 2 issues here. My unrar path syntax may be wrong?, but I cannot change the path as it will not save.

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by sharkykh** [comment]
Please enable debug in general config,
try to set unrar tool location in post-processing config (_**see image for example**_)
Then go to View Log, change Level to `Debug`, copy the logs and upload them using [pastebin](http://pastebin.com) and post the link here.

![image](https://cloud.githubusercontent.com/assets/10238474/23654764/e03c6994-033a-11e7-9463-5f9d6d1c595b.png)

**Comment by Quiksmage** [comment]
This is what shows up when I attempt to change the unrar location to "unrar"
It is still not getting saved and reverting to the following "C:\Program Files\WinRAR\UnRAR.exe"
Looking at your example, it appears that I put it in correctly originally going to the direct exe path, so I may not have to change it, but I guess it should still be noted that it isn't saving.

Here is every line that appears in debug after I changed the unrar location and click "save changes".
http://pastebin.com/xJW3axFm

**Comment by sharkykh** [comment]
Ok that log sadly didn't help me.
Again with debug option on, try post processing so that yout get this warning again, and post the debug logs after that please

**Comment by Quiksmage** [comment]
Another note,

I received this message at 4:30:03 today
```
2017-03-07 04:30:03 POSTPROCESSOR-AUTO :: Failed to extract the archive shadowhunters.s02e10.1080p.webrip.x264-deflate.rar: Invalid Rar Archive Usage
```

However, when I look at history, I see that the episode correctly "downloaded" at 3/7/2017, 04:32:29 AM.

Is it possible that although it is giving me a rar error, it is still actually working? I did find an unpacked folder in the directory (which I believe should have been deleted as well after post-processing).


Manually re-post-processing the file over the same file yielded no error as well as an unpacked folder in the rar-folder directory (again, I believe this is supposed to be deleted).

I will now attempt to delete the file from its folder, refresh the show, and change the name of the folder so it hopefully automatically post-processes this episode as if it was newly downloaded.

**Comment by sharkykh** [comment]
about the rar contents not being deleted - if you're not using the 'move' method, you **need to select the option to delete RAR contents** in order for the post-processor to delete the extracted folders after the episode is processed.

**Comment by Quiksmage** [comment]
Yes, that option is checked and has been which is why I'm wondering why it isn't deleting these contents. I'm still working on getting my test from, above working. It may be a few hours till I can confirm the error still happens. Sorry for the wait!

http://i.imgur.com/3NSqjNE.png

**Comment by sharkykh** [comment]
>Yes, that option is checked and has been which is why I'm wondering why it isn't deleting these contents. 

Currently without a complete log I can't know for sure, and I don't remember the code by heart,
But I think if it fails to extract the archive, it won't delete the contents.

**Comment by Quiksmage** [comment]
http://pastebin.com/tAkqv6Vt

'''
2017-03-07 12:44:03 INFO     POSTPROCESSOR-AUTO :: shadowhunters.s02e10.1080p.webrip.x264-deflate.rar : Unpacking failed: Unpacking Failed with Invalid Rar Archive Usage. Is unrar installed and on the system PATH?
'''

unrar definitely is at C:\Program Files\WinRAR\UnRAR.exe for reference. I'd love to change it back to just unrar since I have the system variable in place now. Is there a manual file I can change this in since my web saving is not working on this page.

**Comment by sharkykh** [comment]
Yes of course.
* Shutdown SickRage.
* In your data directory you have config.ini - edit the file:
  * Find: `unrar_tool = C:\Program Files\WinRAR\UnRAR.exe`
  * Change to `unrar_tool = unrar`
* Start SickRage.
* Check post-processing config to see that it changed.
* Run post-processing again.

**Comment by Quiksmage** [comment]
changed back to to "unrar"

I now do not receive the unpacking failed error. I'm unsure why my path directly to C:\Program Files\WinRAR\UnRAR.exe would fail.

However, the contents of the unpack are still left behind.

I also tested changing the unrar path again and saving it. This time it properly saved. I was also able to again change it to "unrar" and save this as well. It appears as though my save problem has disappeared.

So, now the only problem left is why "C:\Program Files\WinRAR\UnRAR.exe" yields and error, but still unpacks, and why the unpacked contents are left behind even without an error or warning.

**Comment by sharkykh** [comment]
Okay so let's recap:
- [x] Problem saving the post-processing config

- [ ] "C:\Program Files\WinRAR\UnRAR.exe" extracts RARs but yields an error
**I don't know how to reproduce this issue**

- [ ] RAR contents not deleted
It works for most people - post processing is one of the most worked on sections of SickRage.
**So, let's try something else:**
  - post a photo of your post-processing settings
  - a list of the download folder and everything in it (you can do that in cmd
with `tree <path> /F /A`

  And I'll try to find the reason why it's not deleting, or try to reproduce this issue

**Comment by Quiksmage** [comment]
**RAR contents not deleted.**

This used to work fine for me before all of the winrar changes. I'm not sure when it stopped working exactly or if all of my rars have this same problem when unpacked. I rarely get rar releases as my primary websites do not allow them.

**Post-Processing settings images**
http://i.imgur.com/nMAjKncg.png
http://i.imgur.com/HGP4inB.png

**Tree Contents**
http://i.imgur.com/u7iW6o6.png

**Comment by sharkykh** [comment]
I was able to reproduce this issue (RAR contents not deleted when method is not 'move' and delete contents is selected)
I will push an update that will fix this (updates are usually merged within a few days)

Thank you too :)

**Comment by sharkykh** [comment]
https://github.com/SickRage/SickRage/pull/3358 will fix that issue.
